### PR TITLE
Feature/categories interests

### PR DESCRIPTION
**Pull request by GhaliBerbich**
---

### Overview
This PR refines our approach to managing user interests within the application. By centralizing this functionality in the `ManageInterests` function, we ensure uniform behavior across both the `ProfileCreation.kt` and `EditProfile.kt` screens. The introduction of a structured data class for interests further enhances our system's capability to efficiently manage and utilize user data.

### Features Implemented
- **ManageInterests Function**: A new, centralized function for adding and managing interests which requires users to select a category before they can add an interest. This function is utilized across multiple user profile management screens to maintain consistency.
- **Data Class for Interests**: Implementation of a new data class for interests that encapsulates a category and an interest value, allowing for more structured data handling and future functionality enhancements such as suggesting targeted activities for users.
- **Adjustments in Data Handling**: Modifications in how user interests are parsed from Firestore documents to align with the new data structure. Adjustments to function parameters to accommodate these changes have also been made.

### Test Scenarios and Their Descriptions
Comprehensive testing has been conducted to ensure the functionality behaves as expected under various conditions:

1. **Add Interest Button Functionality**:
    - Validates that after selecting a valid category, users can input an interest, and upon submission, the interest appears in the list.

2. **Cannot Write Interest Before Choosing Category**:
    - Confirms that the interest input field remains disabled until a valid category is selected, preventing premature data entry.

3. **Add Button Enabled Only With Both Fields Filled**:
    - Ensures that the add button remains disabled until both the category and interest fields are properly filled, enhancing form validation.

4. **Add Interest Displays It In List**:
    - Checks that once an interest is added, it is correctly displayed in the user's interest list, confirming UI updates.

5. **Deleting An Interest**:
    - Verifies the functionality for removing an interest. Ensures that once an interest is deleted, it is no longer displayed in the list, demonstrating dynamic UI updates.

6. **Add Button Disabled When Category Is None**:
    - Tests that the add button and interest input field are disabled when the category is set to "None", ensuring that invalid data cannot be added.

7. **Prevent Adding When Category Is None Even If There Is An Existing Interest**:
    - Further validates that changing the category to "None" after entering an interest ensures the add button remains disabled.

8. **All Categories Appear OnClick**:
    - Confirms that all predefined categories, including the "None" option, appear correctly when the category dropdown is expanded.

### Conclusion
These updates not only improve the user interface for managing interests but also prepare our system for future enhancements such as personalized activity suggestions based on categorized interests. The extensive tests implemented ensure the reliability of the new functionalities under various scenarios.

---

### CODE REVIEW

**Comment by GhaliBerbich** [comment]
Removed a test for deleting an interest, and a test for adding an interest(which is already tested in other tests), because they would not pass on the CI, eventhough I made the changes to make them pass on small phone. Anyways, line coverage is still 94% on new code, so I will get back to this if I have time later.

File: app/src/main/java/com/android/sample/model/profile/ProfilesRepositoryFirestore.kt [hunk 1]
```diff
@@ -35,7 +35,18 @@ open class ProfilesRepositoryFirestore @Inject constructor(private val db: Fireb
...
+                (interestData as? Map<*, *>)?.let {
+                  val category = it["category"] as? String
+                  val interest = it["interest"] as? String
+                  if (category != null && interest != null) {
+                    Interest(category, interest)
+                  } else {
+                    null
+                  }
+                }
+              },
```
Comment (aminebenji04) [inline app/src/main/java/com/android/sample/model/profile/ProfilesRepositoryFirestore.kt:49]: good handling of the new dataclass, same in ActivityRepositoryFirestore

File: app/src/androidTest/java/com/android/sample/ui/profile/EditProfileTest.kt [hunk 2]
```diff
@@ -92,9 +96,104 @@ class EditProfileScreenTest {
...
+
+    // "Add" button should still be disabled because the interest field is empty.
+    composeTestRule.onNodeWithTag("addInterestButton").assertIsNotEnabled()
+
+    // Enter text in the interest input
+    composeTestRule.onNodeWithTag("newInterestInput").performTextInput("Football")
+
+    // Now, "Add" button should be enabled as both fields are correctly filled.
+    composeTestRule.onNodeWithTag("addInterestButton").assertIsEnabled()
+  }
```
Comment (aminebenji04) [inline app/src/androidTest/java/com/android/sample/ui/profile/EditProfileTest.kt:149]: this is very important

**Review by aminebenji04 — APPROVED** [review]
LGTM! This is a very good PR, new additions are very functional and your tests are good, and the coverage is good also. Good job!

**Pull request merged**